chains/btc/chain.go

```diff
@@ -0,0 +1,94 @@
+// The Licensed Work is (c) 2022 Sygma
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package btc
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/btcsuite/btcd/rpcclient"
+
+	"github.com/ChainSafe/sygma-relayer/chains"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/sygmaprotocol/sygma-core/relayer/message"
+	"github.com/sygmaprotocol/sygma-core/relayer/proposal"
+	"github.com/sygmaprotocol/sygma-core/store"
+)
+
+type BatchProposalExecutor interface {
+	Execute(msgs []*message.Message) error
+}
+type EventListener interface {
+	ListenToEvents(ctx context.Context, startBlock *big.Int, domainID uint8, blockstore store.BlockStore)
+}
+type BtcChain struct {
+	connection *rpcclient.Client
+	listener   EventListener
+	blockstore *store.BlockStore
+	config     *BtcConfig
+	logger     zerolog.Logger
+}
+
+func NewBtcChain(
+	connection *rpcclient.Client, listener EventListener,
+	blockstore *store.BlockStore, config *BtcConfig,
+) *BtcChain {
+	return &BtcChain{
+		connection: connection,
+		listener:   listener,
+		blockstore: blockstore,
+		config:     config,
+		logger:     log.With().Str("domainID", string(*config.GeneralChainConfig.Id)).Logger()}
+}
+
+func (c *BtcChain) Write(msgs []*proposal.Proposal) error {
+	return nil
+}
+
+// remove this after
+func (c *BtcChain) ReceiveMessage(m *message.Message)/* *proposal.Proposal, error */ {}
+
+func (c *BtcChain) PollEvents(ctx context.Context) {
```

Comment:
I think is now changed to accept start block so this logic should be moved to the app.

Reply:
Look at EVM chain.go